src/test/java/io/lettuce/core/cluster/pubsub/RedisClusterPubSubConnectionIntegrationTests.java

```diff
@@ -126,13 +133,55 @@ void subscribeToShardChannelViaReplica() {
         int clusterKeyslot = connection.sync().clusterKeyslot(shardChannel).intValue();
         String thisNode = connection.getPartitions().getPartitionBySlot(clusterKeyslot).getNodeId();
 
-        RedisPubSubAsyncCommands<String, String> replica =
-                pubSubConnection.async().nodes(node -> thisNode.equals(node.getSlaveOf())).commands(0);
+        RedisPubSubAsyncCommands<String, String> replica = pubSubConnection.async()
+                .nodes(node -> thisNode.equals(node.getSlaveOf())).commands(0);
         replica.ssubscribe(shardChannel);
 
         Wait.untilEquals(shardChannel, connectionListener.getShardChannels()::poll).waitOrTimeout();
     }
 
+    @Test
+    @EnabledOnCommand("SSUBSCRIBE")
+    void publishToShardChannel() throws Exception {
+        pubSubConnection.addListener(connectionListener);
+        pubSubConnection.async().ssubscribe(shardChannel);
+        Wait.untilEquals(shardChannel, connectionListener.getShardChannels()::poll).waitOrTimeout();
+
+        pubSubConnection.async().spublish(shardChannel, shardMessage);
+        Wait.untilEquals(shardChannel, connectionListener.getShardChannels()::poll).waitOrTimeout();
+        Wait.untilEquals(shardMessage, connectionListener.getMessages()::poll).waitOrTimeout();
```

Comment:
Do we want to also verify the message and channel name are correct?

A negative test could verify that the message is not received after being published if we are not subscribed to the specific shard channel.

Reply:
Yes and no :)

IMHO it would be good for us to have a test that checks the whole flow - from subscription to notification. There could be many things we might do wrong along the way - we can modify the initial message before we send it to the server or we can modify / decode it wrong / when we receive it back.

Even if the server processes the data properly we could still get wrong results.

Reply:
> we can modify the initial message before we send it to the server or we can modify / decode it wrong / when we receive it back.

all those are addressed in the integration tests currently , 
i am happy to add if you can give the specifics of the case.

Reply:
No need to add them as long as they are addressed. Thanks.